src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1510,6 +1458,24 @@ else if (ticket == ChunkTicketType.POST_TELEPORT) // post teleport
                 cc.s.getWorld().getChunkManager().addTicket(ChunkTicketType.UNKNOWN, target, radius, target);
             return new NumericValue(ticket.getExpiryTicks());
         });
+    }
+
+    @ScarpetFunction(maxParams = -1)
+    public Value query_noise(Context c, int x, int y, int z, String... noiseQueries) {
```

Comment:
Not sure this gives you ability to run query_noise(block), or you always need to provide x, y and z

Reply:
Using `@Locator.Block BlockPos block` instead of the three ints should allow that.

Reply:
I replaced it with altisi's suggestion, doing testing rn